backend/src/api/mutation.rs

```diff
@@ -162,4 +164,54 @@ impl Mutation {
     async fn remove_block(id: Id, context: &Context) -> ApiResult<RemovedBlock> {
         BlockValue::remove(id, context).await
     }
+
+    /// Atomically mount a series into an (empty) realm;
+    /// Creates all the necessary realms on the path to the target
+    /// and adds a block with the given series at the leaf.
+    async fn mount_series(
+        oc_series_id: String,
+        parent_realm_path: String,
+        #[graphql(default = vec![])]
+        new_realms: Vec<RealmSpecifier>,
+        context: &Context
+    ) -> ApiResult<Realm> {
+
+        let parent_realm = Realm::load_by_path(parent_realm_path, context)
+            .await?
+            .ok_or_else(|| invalid_input!("`parentRealmPath` does not refer to a valid realm"))?;
+
+        if new_realms.is_empty() {
+            let blocks = BlockValue::load_for_realm(parent_realm.key, context).await?;
+            if !blocks.is_empty() {
+                return Err(invalid_input!("series can only be mounted in empty realms"));
+            }
+        }
+
+        let target_realm = {
+            let mut target_realm = parent_realm;
+            for RealmSpecifier { name, path_segment } in new_realms {
+                target_realm = Realm::add(NewRealm {
+                    name,
+                    path_segment,
+                    parent: Id::realm(target_realm.key),
+                }, context).await?
+            }
+            target_realm
+        };
+
+        let series = Series::load_by_opencast_id(oc_series_id, context)
+            .await?
+            .ok_or_else(|| invalid_input!("`oc_series_id` does not refer to a valid Opencast series ID"))?;
```

Comment:
I know it's all in a transaction, but intuitively I would put this further up, before any mutations. :shrug: 

Also, what casing do we use in error messages? The field in the API is called `ocSeriesId`, so maybe we used camel case in these error messages?

Reply:
Hmm I put it there because I only need it there. The code above is purely about realm creation. Putting getting the series up there kind of smells like "declare all your variables at the top" to me. :thinking: 

The casing is a valid point tho, will fix.

Reply:
It's more like "check all your inputs at the top" bc the series id and parent path are the two main things that could be "wrong input". But whatever, as I said.

Reply:
Fair enough, actually. :thinking: